Source/7_Utils/UtilityClasses/EventsProcessor.cs

```diff
@@ -1,28 +1,31 @@
 ﻿using System;
 using System.Collections.Generic;
 using BeatLeader.Models;
+using BeatLeader.Models.AbstractReplay;
 
 namespace BeatLeader {
     public class EventsProcessor<T> : IEventsProcessor where T : struct {
         public delegate float TimeSelectorDelegate(T item);
 
-        public EventsProcessor(IEnumerable<T> events, TimeSelectorDelegate timeSelector) {
+        public EventsProcessor(IEnumerable<T> events, IReplayNoteComparator noteComparator, TimeSelectorDelegate timeSelector) {
```

Comment:
EventsProcessor is type-agnostic. IReplayNoteComparator is not needed here, you can populate it in event-specific processors.